src/manager/msNotificationManager.ts

```diff
@@ -28,18 +31,24 @@ export class MsNotificationManager extends MsManager {
     }
     check() {
         if (getSettings('tweaks').get_boolean('disable-notifications')) return;
+        let uuid = Me.stateManager.getState('notification-uuid');
+        if (!uuid) {
+            uuid = GLib.uuid_string_random();
+            Me.stateManager.setState('notification-uuid', uuid);
+        }
+
         const previousCheck = Me.stateManager.getState('notification-check')
             ? new Date(Me.stateManager.getState('notification-check'))
             : new Date();
 
         const message = Soup.Message.new(
             'GET',
-            `${API_SERVER}/notifications?lastCheck=${previousCheck.toISOString()}`
+            `${API_SERVER}/notifications?lastCheck=${previousCheck.toISOString()}&uuid=${uuid}&gnomeVersion=${PACKAGE_VERSION}`
```

Comment:
Is it not reasonable to send material shell version too? While you are at it?

Reply:
We don't really have a proper Material Shell version  :D

Reply:
ok done !